sdk/cognitiveservices/azure-ai-inkrecognizer/azure/ai/inkrecognizer/_client.py

```diff
@@ -0,0 +1,324 @@
+    value = kwargs.pop(param_name, None)
+    if value is None:
+        return default
+
+    _validate_param(value, param_name, types)
+    return value
+
+
+class _AzureConfiguration(Configuration):
+    def __init__(self, credential, **kwargs):  # pylint:disable=super-init-not-called
+        self._set_universal(**kwargs)
+        # sync-specific azure pipeline policies
+        scopes = []
+        if "scopes" in kwargs:
+            scopes = kwargs.pop("scopes")
+        self.authentication_policy = BearerTokenCredentialPolicy(credential, *scopes, **kwargs)
+        self.retry_policy = kwargs.get("retry_policy", RetryPolicy(**kwargs))
+        self.redirect_policy = kwargs.get("redirect_policy", RedirectPolicy(**kwargs))
+        self.transport = kwargs.get("transport", RequestsTransport())
+
+    def _set_universal(self, **kwargs):
+        Configuration.__init__(self)
+        self.connection = kwargs.get("connection_configuration", ConnectionConfiguration(**kwargs))
+        self.user_agent_policy = kwargs.get("user_agent_policy", UserAgentPolicy(**kwargs))
+        self.proxy_policy = kwargs.get("proxy_policy", ProxyPolicy(**kwargs))
+        self.logging_policy = kwargs.get("logging_policy", NetworkTraceLoggingPolicy(**kwargs))
+        self.headers_policy = kwargs.get("headers_policy", HeadersPolicy(**kwargs))
+
+
+class _InkRecognizerConfiguration():
+    def __init__(self, **kwargs):
+        # Ink Recognizer request arguments
+        self.application_kind = _get_and_validate_param(kwargs, "application_kind", ApplicationKind)
+        self.ink_point_unit = _get_and_validate_param(kwargs, "ink_point_unit", InkPointUnit)
+        self.language = _get_and_validate_param(kwargs, "language", str)
+        self.unit_multiple = _get_and_validate_param(kwargs, "unit_multiple", (int, float))
+        self.service_version = _get_and_validate_param(kwargs, "service_version", ServiceVersion)
+        # azure service common arguments
+        self.response_hook = kwargs.get("response_hook", None)
+        if self.response_hook is not None:
+            if not callable(self.response_hook):
+                raise ValueError("response_hook should be callable.")
+        timeout = _get_and_validate_param(kwargs, "timeout", (int, float))
+        client_request_id = _get_and_validate_param(kwargs, "client_request_id", str)
+        self.headers = _get_and_validate_param(kwargs, "headers", dict, {})
+        self.headers["x-ms-client-request-id"] = client_request_id or self._generate_uuid()
+        if timeout is not None:
+            kwargs["connection_timeout"] = timeout
+        self.kwargs = kwargs
+
+    def _generate_uuid(self):  # pylint:disable=no-self-use
+        return str(uuid.uuid4())
+
+
+class _InkRecognizerClientBase:
+    def __init__(self, url, credential, **kwargs):
+        # type: (str, TokenCredential, Any) -> None
+        _validate_param(url, "url", str)
+
+        self._logger = _LOGGER
```

Comment:
No need to have an instance attribute. Just use the module level attribute.

Reply:
Removed